File: cases/typescript_openai_approval_decision_env/agent.ts
import { Agent, run } from "@openai/agents";

const AUTO_APPROVE_HITL = process.env.AUTO_APPROVE_HITL === "1";

async function confirm(question: string): Promise<boolean> {
  if (AUTO_APPROVE_HITL) {
    console.log(`[auto-approve] ${question}`);
    return true;
  }
  return question.length > 0;
}

const agent = new Agent({
  name: "Approval decision agent",
});

let result = await run(agent, "start approval flow");
const approvalState = result.state;
for (const interruption of result.interruptions ?? []) {
  const ok = await confirm("approve?");
  if (ok) {
    approvalState.approve(interruption);
  } else {
    approvalState.reject(interruption);
  }
}

let inlineResult = await run(agent, "start direct approval flow");
for (const interruption of inlineResult.interruptions ?? []) {
  if (await confirm("approve inline?")) {
    inlineResult.state.approve(interruption);
  }
}

let reassignedResult = await run(agent, "start reassigned approval flow");
for (const interruption of reassignedResult.interruptions ?? []) {
  let maybe = await confirm("approve after reassignment?");
  maybe = false;
  if (maybe) {
    reassignedResult.state.approve(interruption);
  }
}

File: cases/python_openai_approval_decision/agent.py
import agents
from agents import Agent, Runner


agent = Agent(name="Python approval decision agent")


async def main() -> None:
    result = await Runner.run(agent, "start approval flow")
    state = result.to_state()
    for interruption in result.interruptions:
        state.approve(interruption)
        state.reject(interruption)

    stream_result = agents.Runner.run_streamed(agent, "start streaming approval flow")
    stream_state = stream_result.to_state()
    for interruption in stream_result.interruptions:
        stream_state.approve(interruption)

    loose_state = object()
    loose_state.approve("not an SDK state")

    rebound_state = result.to_state()
    rebound_state = object()
    rebound_state.approve("not an SDK state after reassignment")
